File: CovidWatch/app/src/main/java/com/riskre/covidwatch/data/ContactEventViewModel.java
package com.riskre.covidwatch.data;


import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

/**
 * TODO
 */
public class ContactEventViewModel extends AndroidViewModel {

    private CovidWatchRepository repository;
    private LiveData<List<ContactEvent>> allEvents;

    /**
     * TODO
     * @param application
     */
    public ContactEventViewModel(Application application) {
        super(application);
        repository = new CovidWatchRepository(application);
        allEvents = repository.getAllEvents();
    }

    /**
     * TODO
     * @return
     */
    public LiveData<List<ContactEvent>> getAllEvents() {
        return allEvents;
    }

}

File: CovidWatch/app/src/main/java/org/covidwatch/android/data/ContactEventViewModel.kt
package org.covidwatch.android.data

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

class ContactEventViewModel(application: Application) : AndroidViewModel(application) {

    private val repository: CovidWatchRepository = CovidWatchRepository(application)

    /**
     * TODO
     * @return
     */
    val allEvents: LiveData<List<ContactEvent>>

    /**
     * TODO
     * @param application
     */
    init {
        allEvents = repository.allEvents
    }
}